src/stdlib/mod.rs

```diff
@@ -163,3 +169,127 @@ fn assert<B: Backend>(
 
     Ok(None)
 }
+
+fn to_bits<B: Backend>(
+    compiler: &mut CircuitWriter<B>,
+    generics: &GenericParameters,
+    vars: &[VarInfo<B::Field, B::Var>],
+    span: Span,
+) -> Result<Option<Var<B::Field, B::Var>>> {
+    // should be two input vars
+    assert_eq!(vars.len(), 2);
+
+    // but the better practice would be to retrieve the value from the generics
+    let num_var = generics.get("LEN");
+
+    // num should be greater than 0
+    assert!(num_var > 0);
```

Comment:
let's also assert that `bitlen <= MODULUS.bitlen() - 1`

Reply:
Added.
But why it is not `bitlen <= MODULUS.bitlen()`?

Reply:
Imagine a modulus that’s not 1111, but like 1001 for example

Reply:
so if the modulus is 9 (1001), the `MODULUS.bitlen() - 1` will only allow it represent max value as 111, which is 7.

then it doesn't work for the value 8?

Reply:
how would you prove that like 10, 11, 12, 13, 14, and 15 are correctly constrained?